packages/dev/core/src/Misc/decorators.ts

```diff
@@ -24,8 +24,8 @@ function generateExpandMember(setCallback: string, targetKey: Nullable<string> =
             set: function (this: any, value) {
                 // does this object (i.e. vector3) has an equals function? use it!
                 // Note - not using "with epsilon" here, it is expected te behave like the internal cache does.
-                if (typeof this.equals === "function") {
-                    if (this.equals(value)) {
+                if (typeof this[key].equals === "function") {
+                    if (this[key].equals(value)) {

```

Comment:
this is incorrect, isn't it? this is not the function context. it is the object on which the property is set.

Reply:
Yes, line 32 shows the code here is aim to compare `this[key]` and `value`, so "this object" in the comment should means `this[key]`, not `this`.

Reply:
Do you have an example that reproes the issue ?

Reply:
This would help create a test for it.

Reply:
Also I guess the check could be moved outside the DefineProperty to prevent the extra runtime cost on every assignment :-)

Reply:
then it should probably be

```suggestion
                if (typeof this[key]?.equals === "function") {
                    if (this[key].equals(value)) {
```

Reply:
playground: https://playground.babylonjs.com/#OH54TW
I set `_position` and `_isWorldMatrixFrozen` to their default values, function `aaa` should not be called, but actually called when I set `_position`, it is not expected.

Reply:
LGTM!